### ISSUE TITLE
Add an intStream() method on Int collections, longStream() on Long Collections, etc

### ISSUE THREAD

**Issue by techsy730** [vigna/fastutil#116]
Java has primitive stream types for int, long, and double, that let you use the operations of stream without boxing/unboxing. It would be nice for the int, long, and double collections to export a stream that natively works in those types.

I think there should be a way to do this without having to re-implement the entire streaming framework implementation, but if it turns out there isn't, feel free to close this as infeasible.

**Comment by vigna** [comment]
Yeah, that had been on the list for a while. Sorry, i'm just overwhelmed😭

**Comment by techsy730** [comment]
I may have some time to tackle this along with the spliterator thing. Do you mind if I try my hand at it and maybe send you a pull request in a week or two?

**Comment by vigna** [comment]
> On 7 Mar 2019, at 10:08, TechSY730 <notifications@github.com> wrote:
> 
> I may have some time to tackle this along with the spliterator thing. Do you mind if I try my hand at it and maybe send you a pull request in a week or two?

Go for it!

Ciao,

					seba

**Comment by techsy730** [comment]
I've made a bunch of progress on this. I have added default Spliterator methods for Collection, List, Set, and SortedSet. I have also added a specific implementation for ArrayList, as will as intStream, longStream, and doubleStream as appropriate to the type-specific interface.
While I was at it, I also added a bunch of new default methods (like sort on the list types that delegate to the type-specific array sort in XArrays). If you are not comfortable with that, I can carve them out.

I'm not ready for a pull request yet. I still want to add Spliterator implementations for, at minumum, ArraySet, ArrayMap, BigList, and BigArrayList. If I can get to the Hash based ones, that would be nice too, but those are a bit trickier. And of course I need to still test the dang things. And I got the version code messed up (I annotated it @since 8.1.0, but I see now we are already on the 8.2s), but just wanted to let you know it is coming along.

https://github.com/techsy730/fastutil/tree/spliterators
